scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py

```diff
@@ -131,21 +139,38 @@ def __init__(self, args: dict):
         self._report_file = PurePath(self._report_workspace).joinpath(
             f"report.{self._instance_id}.json")
         args[ARG_REPORT] = self._report_file
-        self._parse_arguments_to_runtime(
-            args, self._runtime_args)
+        self._parse_arguments_to_runtime(args)
+
+    def _compile_multi_value_arg_string(self, multi_values):
+        """
+        Takes the multi values of a command line argument and returns a string with each value separated by a dash character.
+
+        @param multi_args: The multi values of the argument to compile as a dash-separated string.
+        """
 
-    def _parse_arguments_to_runtime(self, args, runtime_args):
+        num_values = len(multi_values)
+        multi_values_string = ""
+        for i in range(num_values):
+            multi_values_string += multi_values[i]
+            if i < num_values - 1:
+                multi_values_string += "-"
+
+        return multi_values_string
+
+    def _parse_arguments_to_runtime(self, args):
         """
         Fetches the relevant keys from the provided dictionary, and applies the values of the arguments(or applies them as a flag) to our runtime_args list.
 
         @param args: Dictionary containing the arguments passed to this TestImpact object. Will contain all the runtime arguments we need to apply.
-        @runtime_args: A list of strings that will become the arguments for our runtime.
         """
 
         for argument in RuntimeArgs:
             value = args.get(argument.driver_argument)
             if value:
-                runtime_args.append(f"{argument.runtime_arg}{value}")
+                if type(value) == list:
+                    self._runtime_args.append(f"{argument.runtime_arg}{','.join(value)}")
+                else:
+                    self._runtime_args.append(f"{argument.runtime_arg}{value}")
```

Comment:
The `type` keyword shouldn't be used in this case as it would form an exact match, but instead the isintance keyword should be used as that is more [pythonic](https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance).
```suggestion
                self._runtime_args.append(f"{argument.runtime_arg}{value if isinstance(value, str) else ','.join(value)}") 
```

Reply:
Fixed 👍